### ISSUE TITLE
Generating p2 site locally

### ISSUE THREAD

**Issue by jmarreroc** [CMark/osgi-swagger-connector#2]
Hi Mark,

I'm trying to generate a new targetplatform with the swagger version 1.5.2-M1 (instead of 1.3.10), and changed the pom properly. But when I launch the maven command "mvn clean verify jetty:run -Pregenerate" over the "com.cmark.swagger.dep.p2" project, it seems to work correctly downloading but finally I get an
`'java.lang.ClassNotFoundException:org.eclipse.tycho.core.osgitools.targetplatform.DefaultTargetPlatform'.
`

I just installed the Maven tycho utilities & Build Tools for eclipse, but got the same error... Am I missing something?

Thank you again!

**Comment by jmarreroc** [comment]
Hi Mark,

I was researching about the problem and it seems to be a [reficio p2 maven plugin](https://github.com/reficio/p2-maven-plugin/issues/64) issue when build with the newests versions of tycho (in this case with 0.21.0 or 0.22.0). If you downgrade this version it works.:

```
<tycho.version>0.18.1</tycho.version>
<tycho.extras.version>0.18.1</tycho.extras.version>
```

So, I'm facing another problem when I upgrade the swagger version from _1.3.10_ to _1.5.2-M1_. (I'm trying to upgrade swagger version because it gives support for Jax RS resource locator) I get a non numeric version exception: invalid version "1.5.2-M1": non-numeric "2-M1".
The problem seems to be that the generated version 1.5.2-M1 doesn't adjust to generic bundle versions specification. I trying to configure the tycho-packaging-plugin just to force the bundle versioning with no results. Have you face this before?
Cheers!

**Comment by cmark** [comment]
Hi @jmarreroc, 

Thanks for reporting the issue.
I think you've found the source of the problem.
I'll change the `com.cmark.swagger.dep.p2` module's pom.xml to use `Tycho 0.20.0` until the issue you've referenced is not fixed. Seems to work with `Tycho 0.20.0`, and I think I've used that when generated the initial version of the deployed p2 site, and later decided to bump to `0.21`, but never tried to generate this p2 site again.

About the swagger version bump to `1.5.x`. I've tried it and got the same problem, and unfortunately I haven't seen that exception, until now :smile:. I think the problem is in the Export-Package instructions where I blindly reuse the defined Swagger version (Maven style), which in this case is an incorrect OSGi version, therefore it throws the exception when tries to generate a p2 site.

I pushed my changes to a new `swagger-1.5.x` branch. You can check it out, I haven't tried the new swagger bundles, though.

All the best,
Mark

**Comment by jmarreroc** [comment]
Hi @CMark!

Now the P2 site is generating correctly! But it seems to be several changes from this swagger 1.5.x version, and the publisher need some refactoring :( 
I'm deeping inside your publisher and the swagger 1.5.3-M1 code to solve this issues, but it seems not to be pretty easy.

See you later!

**Comment by cmark** [comment]
Yeah, I know that's why I put the changes to a branch, I'll look into the matter when I have more time.

**Comment by jmarreroc** [comment]
Hi @CMark,
I'm working on your branch, but _tycho+reficio_ is giving me a headache as I'm not an expert with these plugins... I just have included _yaml_ and _scala_ libraries as I see that the new swagger version need them (Maybe there is another automatically way to improve this and inject this dependencies...)
So, now I'm facing another problem maybe related with how tycho manage the import/export packages (but I repeat: I'm totally noob): I'm trying to define the Swagger config during the bundle Activator as you did on your current publisher version. So, I'm defining a swagger _BeanConfig_ programatically following [this example](https://github.com/swagger-api/swagger-core/wiki/Java-CXF-Quickstart#specify-a-swagger-configuration-class). I just not consider to use DefaultJaxrsConfig because this one uses web.xml for Servlet specification. 

So, the problem i'm facing is when I create the _BeanConfig_ instance: For some reason cannot find the _Scanner_ interface who implements... I see the _com.wordnik.swagger-jaxrs.1.5.3.M1.jar_ manifest generated but I can't find the **com.wordnik.swagger.config.**\* import I specified in the p2 pom... I see that the **com.wordnik.swagger.config.**\* package exists both _com.wordnik:swagger-jaxrs_ and _com.wordnik:swagger-core_ and don't know if exactly this could be the problem as tycho doesn't know how to resolve the package **com.wordnik.swagger.config.**\* and import only the first package encountered. Do you know what I'm doing wrong?

Cheers!

**Comment by jmarreroc** [comment]
It seems that my user has no permissions to commit on your branch. If you want i can commit my few changes

**Comment by jmarreroc** [comment]
I have discovered something important relevant to subresource locator support on swagger 1.3.x and the current swagger osgi connector (around the SwaggerResourceTracker code specification). If you consider it right I can create another issue about this respect. The problem is quite simple in its resolution, but anyway subresource locator support with swagger 1.3.x is quite inmature and don't support some resource specifications.
